### PR TITLE
Reworked models of vc-module-quotes

### PR DESCRIPTION
**Pull request by MalikJuly**

### CODE REVIEW

File: VirtoCommerce.QuoteModule.Web/Converters/TierPriceConverter.cs [hunk 1]
```diff
@@ -1,11 +1,6 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Web;
 using Omu.ValueInjecter;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/TierPriceConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Web/Converters/QuoteRequestTotalsConverter.cs [hunk 2]
```diff
@@ -1,11 +1,6 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Web;
 using Omu.ValueInjecter;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/QuoteRequestTotalsConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Web/Converters/QuoteRequestConverter.cs [hunk 3]
```diff
@@ -1,15 +1,11 @@
-﻿using System;
-using System.Collections.Generic;
 using System.Linq;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/QuoteRequestConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Web/Converters/QuoteItemConverter.cs [hunk 4]
```diff
@@ -1,11 +1,7 @@
-﻿using System;
-using System.Collections.Generic;
 using System.Linq;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/QuoteItemConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Web/Converters/QuoteAttachmentConverter.cs [hunk 5]
```diff
@@ -1,11 +1,6 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Web;
 using Omu.ValueInjecter;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/QuoteAttachmentConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Web/Converters/AddressConverter.cs [hunk 6]
```diff
@@ -1,15 +1,10 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Web;
 using Omu.ValueInjecter;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Web/Converters/AddressConverter.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Test/QuoteModuleTests.cs [hunk 7]
```diff
@@ -0,0 +1,408 @@
+using System;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Test/QuoteModuleTests.cs]: Not sure if we need to add this test at all. There is no much sense in adding such conversion tests. 
I would add test for TotalCalculators calculation logic. Maybe data prepared for TotalCalculator_ReturnsQuoteRequestTotals could be used for this.

File: VirtoCommerce.QuoteModule.Data/Services/DefaultQuoteTotalsCalculator.cs [hunk 8]
```diff
@@ -1,18 +1,13 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿using System.Linq;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Services/DefaultQuoteTotalsCalculator.cs]: No significant changes, can be removed from commit

File: VirtoCommerce.QuoteModule.Data/Converters/AddressComparer.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Converters/AddressComparer.cs]: Do we need it - is it used somewhere?

File: VirtoCommerce.QuoteModule.Data/Model/AddressEntity.cs [hunk 10]
```diff
@@ -43,8 +40,75 @@ public class AddressEntity : Entity
...
+        public virtual QuoteRequestEntity QuoteRequest { get; set; }
+        
+        #endregion
+       
+        public virtual Address ToModel(Address address)
+        {
+            if (address == null)
+              throw new ArgumentNullException(nameof(address));
+
+            address.AddressType = (AddressType)Enum.Parse(typeof(AddressType), this.AddressType);
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Model/AddressEntity.cs]: Why InjectFrom is not used for simple type properties with the same names?

File: VirtoCommerce.QuoteModule.Data/Model/AttachmentEntity.cs [hunk 11]
```diff
@@ -1,35 +1,79 @@
...
+        public virtual QuoteRequestEntity QuoteRequest { get; set; }
+
+        #endregion
+       
+        public virtual QuoteAttachment ToModel(QuoteAttachment attachment)
+        {
+          if (attachment == null)
+            throw new ArgumentNullException(nameof(attachment));
+
+          attachment.Id = this.Id;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Model/AttachmentEntity.cs]: Why InjectFrom is not used for simple type properties with the same names?

File: VirtoCommerce.QuoteModule.Data/Model/QuoteItemEntity.cs [hunk 12]
```diff
@@ -20,45 +18,109 @@ public QuoteItemEntity()
...
-		#endregion
-	}
+       #endregion
+     
+       public virtual QuoteItem ToModel(QuoteItem quoteItem)
+       {
+         if (quoteItem == null)
+           throw new ArgumentNullException(nameof(quoteItem));
+
+         quoteItem.Id = this.Id;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Model/QuoteItemEntity.cs]: Why InjectFrom is not used for simple type properties with the same names?

File: VirtoCommerce.QuoteModule.Data/Model/QuoteRequestEntity.cs [hunk 13]
```diff
@@ -1,98 +1,255 @@
...
+		public virtual ObservableCollection<AttachmentEntity> Attachments { get; set; }
+
+        #endregion
+
+        public virtual QuoteRequest ToModel(QuoteRequest quoteRequest)
+        {
+          if (quoteRequest == null)
+            throw new ArgumentNullException(nameof(quoteRequest));
+       
+          quoteRequest.Id = this.Id;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Model/QuoteRequestEntity.cs]: Why InjectFrom is not used for simple type properties with the same names?

File: VirtoCommerce.QuoteModule.Data/Model/TierPriceEntity.cs [hunk 14]
```diff
@@ -1,25 +1,49 @@
...
-		#endregion
-	}
+        #endregion
+       
+        public virtual TierPrice ToModel(TierPrice tierPrice)
+        {
+          if (tierPrice == null)
+            throw new ArgumentNullException(nameof(tierPrice));
+       
+          tierPrice.Price = this.Price;
```
Comment (yecli) [inline VirtoCommerce.QuoteModule.Data/Model/TierPriceEntity.cs]: Why InjectFrom is not used for simple type properties with the same names?

**Comment by tatarincev** [comment]
Need to rework from scratch.

**Pull request closed without merging**